llvm/lib/Target/RISCV/RISCVRegisterInfo.cpp

```diff
@@ -803,6 +803,11 @@ RISCVRegisterInfo::getRegisterCostTableIndex(const MachineFunction &MF) const {
              : 0;
 }
 
+unsigned
+RISCVRegisterInfo::getSpillWeightFactor(const TargetRegisterClass *RC) const {
+  return getRegClassWeight(RC).RegWeight;
```

Comment:
Why does this need to be a virtual function if it's just reading a property out of the TargetRegisterClass

Reply:
Because I don't know if all targets should use this, so I made it a target hook and overrided it for RISC-V.

Reply:
Do not do this. Just use the field

Reply:
You mean I should apply this change to all the targets (the shared part)?

Reply:
Yes. The opt-out is just use a factor of 1

Reply:
The change will be huge if this is applied to all targets, and targets may have different factors. So I think we can keep the hook and let the targe make the decision.

Reply:
No, do not increase the target API surface out of fear of changing other targets. This is a large problem for llvm where everyone just adds some new single purpose knob that nothing else will ever implement. There must be attempts at universal infrastructure improvements. The targets may have different weights, but there is already the mechanism here to control it. Just use it instead of adding another undiscoverable, redundant override to implement

Reply:
Done.